pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -0,0 +1,56 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package realizestate
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+type RealizeStateService struct {
+	common.Service
+}
+
+type RealizeError struct {
+}
+
+func InitializeRealizeState(service common.Service) *RealizeStateService {
+	return &RealizeStateService{
+		Service: service,
+	}
+}
+
+func IsRealizeStateError(err error) bool {
+	return err.Error() == model.GenericPolicyRealizedResource_STATE_ERROR
+}
+
+// CheckRealizeState allows the caller to check realize status of entityType with retries.
+// backoff defines the maximum retries and the wait interval between two retries.
+func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, orgID, projectID, intentPath, entityType string) error {
```

Comment:
Is it possible to simplify the function from `func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, orgID, projectID, intentPath, entityType string) error {...}` to `func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, intentPath string) error {...}`? i.e.
1. get the org/project id from intentPath instead of parameters
2. entity type is not required, right?

Reply:
> Is it possible to simplify the function from `func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, orgID, projectID, intentPath, entityType string) error {...}` to `func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, intentPath string) error {...}`? i.e.
> 
> 1. get the org/project id from intentPath instead of parameters
> 2. entity type is not required, right?

For 1, I will rebase and update patch after we merged the patch of parsing parameters from path.

For 2, entity type is required in Line 46, return value of `RealizedEntitiesClient.List` is a list, containing realized status of all related intent types. So `entity type` is used to check the status of entity which we really care.

Reply:
Done.